### PR TITLE
fix: next pages strict mode and options

### PR DESCRIPTION
**Pull request by ajwootto**
- remove the "synchronization" logic from the pages router SDK because the SSR data isn't "reactive", it just gets sent once on the client
- just pass it in to the initialization of the DevCycle SDK

### CODE REVIEW

File: sdk/nextjs/src/pages/appWithDevCycle.tsx [hunk 1]
```diff
@@ -87,21 +39,25 @@ export const appWithDevCycle = <Props extends NextJsAppProps>(
             <DevCycleProvider
                 config={{
                     sdkKey: devcycleSSR.sdkKey,
+                    user: devcycleSSR.user,
                     options: {
                         ...additionalOptions,
+                        ...(onServerside
```
Comment (jsalaber) [inline sdk/nextjs/src/pages/appWithDevCycle.tsx:45]: can this be on top of additional options? if the user defines the `disableAutomaticEventLogging` option it'll get overwritten here

Comment (ajwootto) [reply sdk/nextjs/src/pages/appWithDevCycle.tsx:45]: it has to be overwritten on the serverside, because event logging there isn't supported

Comment (jsalaber) [reply sdk/nextjs/src/pages/appWithDevCycle.tsx:45]: ah okay I see it won't override it if the onServerSide is false on the client now

**Review by jsalaber — APPROVED** [review]

File: sdk/nextjs/src/pages/appWithDevCycle.tsx (hunk 1 above)
Comment (ajwootto) [reply sdk/nextjs/src/pages/appWithDevCycle.tsx:45]: yup

**Pull request merged**